modules/transactionpool/update.go

```diff
@@ -8,6 +8,124 @@ import (
 	"github.com/NebulousLabs/Sia/types"
 )
 
+// findSets takes a bunch of transactions (presumably from a block) and finds
+// all of the separate transaction sets within it. Set does not check for
+// conflicts.
+func findSets(ts []types.Transaction) [][]types.Transaction {
```

Comment:
Ok, I can add a comment explaining the algorithm. I should note that the algorithm is not complete yet, and is also not likely to stay in this package as the miner needs to use the exact same algorithm.

I believe that the way it currently works is a graph theory problem, the solution I have here is linear time which is definitely the best O-notation you can get, as you minimally need to look at all the transaction edges anyway.

The extension that needs to happen here is that we need to split off child subsets with lower fees, and that's going to complicate the code a bit. It'll still be linear time though. Basically, instead of greedily combining sets, we'll only combine sets if the combining increases the average fee value of the parent set.